src/Runner.Worker/ActionManager.cs

```diff
@@ -634,6 +641,19 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
             }
         }
 
+        private bool CheckWatermark(string watermarkFile)
+        {
+            if (!File.Exists(watermarkFile))
+            {
+                return false;
+            }
+
+            var content = File.ReadAllText(watermarkFile);
+            return !string.IsNullOrEmpty(content) &&
+                DateTimeOffset.TryParse(content, out DateTimeOffset created) &&
+                DateTimeOffset.Now - created < TimeSpan.FromDays(1);
```

Comment:
i might change this to something very large, like 10 years since this is just a temporary change. I dont really want actions re-downloading after 1 day.

Reply:
how do we handle tag get updates? ex: checkout@v2 suppose to point to latest version of v2 checkout.
i know when you use the api to download the tarball, you will get a nested zip/tar.gz, the root folder is actually the short SHA of the ref it resolve to, i think we can also keep tracking that. So we will not redownload same action only if the underline resolved SHA is not changed between 2 runs.